FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -0,0 +1,428 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Data.Tables;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Synapse.Common.Configurations;
+using Microsoft.Health.Fhir.Synapse.Common.Models.FhirSearch;
+using Microsoft.Health.Fhir.Synapse.Common.Models.Jobs;
+using Microsoft.Health.Fhir.Synapse.Core.DataFilter;
+using Microsoft.Health.Fhir.Synapse.Core.Exceptions;
+using Microsoft.Health.Fhir.Synapse.Core.Extensions;
+using Microsoft.Health.Fhir.Synapse.Core.Fhir;
+using Microsoft.Health.Fhir.Synapse.Core.Jobs.Models;
+using Microsoft.Health.Fhir.Synapse.Core.Jobs.Models.AzureStorage;
+using Microsoft.Health.Fhir.Synapse.DataClient;
+using Microsoft.Health.Fhir.Synapse.DataClient.Api;
+using Microsoft.Health.Fhir.Synapse.DataClient.Extensions;
+using Microsoft.Health.Fhir.Synapse.DataClient.Models.FhirApiOption;
+using Microsoft.Health.Fhir.Synapse.DataWriter;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using JobStatus = Microsoft.Health.JobManagement.JobStatus;
+
+namespace Microsoft.Health.Fhir.Synapse.Core.Jobs
+{
+    public class FhirToDataLakeOrchestratorJob : IJob
+    {
+        private readonly JobInfo _jobInfo;
+        private readonly FhirToDataLakeOrchestratorJobInputData _inputData;
+        private readonly IFhirDataClient _dataClient;
+        private readonly IFhirDataWriter _dataWriter;
+        private readonly IQueueClient _queueClient;
+        private readonly ITypeFilterParser _typeFilterParser;
+        private readonly IGroupMemberExtractor _groupMemberExtractor;
+        private readonly TableClient _metaDataTableClient;
+        private readonly JobSchedulerConfiguration _schedulerConfiguration;
+        private readonly FilterConfiguration _filterConfiguration;
+        private readonly ILogger<FhirToDataLakeOrchestratorJob> _logger;
+
+        private FhirToDataLakeOrchestratorJobResult _result;
+
+        public FhirToDataLakeOrchestratorJob(
+            JobInfo jobInfo,
+            FhirToDataLakeOrchestratorJobInputData inputData,
+            FhirToDataLakeOrchestratorJobResult result,
+            IFhirDataClient dataClient,
+            IFhirDataWriter dataWriter,
+            IQueueClient queueClient,
+            ITypeFilterParser typeFilterParser,
+            IGroupMemberExtractor groupMemberExtractor,
+            TableClient metaDataTableClient,
```

Comment:
We can have a interface for metadata store.

Reply:
Thanks, I will extract and define interfaces for metadata store.

Reply:
Updated.